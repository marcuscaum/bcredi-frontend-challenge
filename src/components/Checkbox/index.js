import React from 'react';
import PropTypes from 'prop-types';

import {
  CheckboxContainer,
  CheckboxInput,
  CheckboxLabel,
} from './index.styled';

const CheckBoxComponent = ({ label, name, fontSize, fontColor, ...props }) => (
  <CheckboxContainer>
    <CheckboxInput name={name} id={`${name}-input-checkbox`} {...props} />
    <CheckboxLabel
      fontColor={fontColor}
      fontSize={fontSize}
      htmlFor={`${name}-input-checkbox`}
    >
      {label}
    </CheckboxLabel>
  </CheckboxContainer>
);

CheckBoxComponent.propTypes = {
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
  name: PropTypes.string.isRequired,
  fontColor: PropTypes.string,
  fontSize: PropTypes.string,
};

CheckBoxComponent.defaultProps = {
  fontSize: '12px',
  fontColor: '#cccccc',
};

export default CheckBoxComponent;
