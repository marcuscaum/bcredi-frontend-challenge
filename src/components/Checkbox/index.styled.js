import styled from 'styled-components';
import CheckImage from 'assets/check.svg';

export const CheckboxLabel = styled.label`
  font-size: ${({ fontSize }) => fontSize};
  color: ${({ color }) => color};
  line-height: 1.67;
  cursor: pointer;
  overflow-wrap: normal;
  float: left;
  margin-left: 35px;

  a {
    text-decoration: none;
    color: ${({ theme: { primary } }) => primary};
  }

  &::before {
    box-sizing: border-box;
    margin: 8px 12px;
    content: '';
    width: 24px;
    height: 24px;
    border-radius: 4px;
    position: absolute;
    background-color: white;
    border: 1px solid #cccccc;
    left: 0;
  }
`;

export const CheckboxInput = styled.input.attrs({
  type: 'checkbox',
})`
  display: none;

  &:checked + ${CheckboxLabel}::before {
    content: url('${CheckImage}');
    background-color: ${({ theme: { primary } }) => primary};
    border: none;
    box-sizing: border-box;
    padding: 2px 7px;
  }
`;

export const CheckboxContainer = styled.div`
  position: relative;
  border: 1px solid ${({ theme: { ultraLightGray } }) => ultraLightGray};
  padding: 10px 12px;
  border-radius: 3px;
`;
