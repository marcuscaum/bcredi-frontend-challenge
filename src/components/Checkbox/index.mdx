---
name: Checkbox
---


import { Playground, PropsTable } from 'docz'
import Checkbox from './index'

# Checkbox

<PropsTable of={Checkbox} />

## Basic usage

<Playground>
  <Checkbox 
    label="Aceito os termos"
    fontSize="22px"
  />
</Playground>