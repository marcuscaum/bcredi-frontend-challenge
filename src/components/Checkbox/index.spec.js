import React from 'react';
import { shallow } from 'enzyme';

import Checkbox from '.';

const setup = (
  props = {
    label: 'Test label',
    name: 'test-name',
  }
) => shallow(<Checkbox {...props} />);

describe('[Component] Checkbox', () => {
  it('should render the component', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
