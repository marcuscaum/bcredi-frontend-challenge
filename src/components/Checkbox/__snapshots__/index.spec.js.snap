// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Checkbox should render the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckBoxComponent
    fontColor="#cccccc"
    fontSize="12px"
    label="Test label"
    name="test-name"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.input
          id="test-name-input-checkbox"
          name="test-name"
/>,
        <styled.label
          fontColor="#cccccc"
          fontSize="12px"
          htmlFor="test-name-input-checkbox"
>
          Test label
</styled.label>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "test-name-input-checkbox",
          "name": "test-name",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Test label",
          "fontColor": "#cccccc",
          "fontSize": "12px",
          "htmlFor": "test-name-input-checkbox",
        },
        "ref": null,
        "rendered": "Test label",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.input
            id="test-name-input-checkbox"
            name="test-name"
/>,
          <styled.label
            fontColor="#cccccc"
            fontSize="12px"
            htmlFor="test-name-input-checkbox"
>
            Test label
</styled.label>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": "test-name-input-checkbox",
            "name": "test-name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Test label",
            "fontColor": "#cccccc",
            "fontSize": "12px",
            "htmlFor": "test-name-input-checkbox",
          },
          "ref": null,
          "rendered": "Test label",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
